### ISSUE TITLE
Sigma crash throwing a std:bad:weak_ptr exception

### ISSUE THREAD

**Issue by Zardoz89** [adam4813/Sigma#159]
I'm trying to build Sigma as STATIC lib and running the executable ( cmake -DBUILD_SHARED_Sigma=OFF -DBUILD_STATIC_Sigma=ON  .. )

The last output it's this : 

```
[INFO] Creating component of type: ALSound
[INFO] Loading Sound from WAV file: audio/trillek_effect_keyboard_spacebar_amp1.wav

terminate called after throwing an instance of 'std::bad_weak_ptr' what():  bad_weak_ptr
```

I compiled it with -DCMAKE_BUILD_TYPE=Debug and run a debugger to trace from were the exception arises. I trace it to the line **191** of **main.cpp** :

```
Sigma::ALSound *als = (Sigma::ALSound  *)alsys.getComponent(200,  Sigma::ALSound::getStaticComponentTypeName());
if(als) {
    als->Play(Sigma::PLAYBACK_LOOP);
}
```

I try it with develop, OpenAL and cleaning-Sigma branch, getting the exact same error.

**Comment by Zardoz89** [comment]
I forgot to say that I run on kUbuntu 13.10 x64

**Comment by Meisaka** [comment]
I added a commit to the OpenAL branch that should show more info in the log or maybe even fix it

**Comment by Zardoz89** [comment]
I keep getting the error, but now there is more useful information. Looks that is related to my problem of not listening any sound inside of Sigma (except the test beep)

```
    [INFO] Loading Sound from WAV file: audio/trillek_effect_spacesuit_breath_amp1.wav
    [WARNING] Failed to load sound from audio/trillek_effect_spacesuit_breath_amp1.wav
    [INFO] Creating component of type: ALSound
    [INFO] Loading Sound from WAV file: audio/trillek_effect_ship_interior_ambient_sdo1.wav
    [WARNING] Failed to load sound from audio/trillek_effect_ship_interior_ambient_sdo1.wav
    [INFO] Creating component of type: ALSound
    [INFO] Loading Sound from WAV file: audio/trillek_effect_ship_interior_ambient_sdo1.wav
    [WARNING] Failed to load sound from audio/trillek_effect_ship_interior_ambient_sdo1.wav
    [INFO] Creating component of type: ALSound
    [INFO] Loading Sound from WAV file: audio/trillek_effect_footsteps_walk_metal_amp1.wav
    [WARNING] Failed to load sound from audio/trillek_effect_footsteps_walk_metal_amp1.wav
    [INFO] Creating component of type: ALSound
    [INFO] Loading Sound from WAV file: audio/trillek_effect_keyboard_alpha_amp1.wav
    [WARNING] Failed to load sound from audio/trillek_effect_keyboard_alpha_amp1.wav
    [INFO] Creating component of type: ALSound
    [INFO] Loading Sound from WAV file: audio/trillek_effect_keyboard_return_amp1.wav
    [WARNING] Failed to load sound from audio/trillek_effect_keyboard_return_amp1.wav
    [INFO] Creating component of type: ALSound
    [INFO] Loading Sound from WAV file: audio/trillek_effect_keyboard_spacebar_amp1.wav
    [WARNING] Failed to load sound from audio/trillek_effect_keyboard_spacebar_amp1.wav
    [WARNING] Invalid Sound index: 0
    terminate called after throwing an instance of 'std::bad_weak_ptr'
      what():  bad_weak_ptr
```

And I have these files, and I can play it with VLC and Audacity....

**Comment by Zardoz89** [comment]
Keep happening, and now It's happens with Shared build of Sigma .... :(

**Comment by adam4813** [comment]
This should be fixed in the latest PR #163
